primitives/src/targeting/eval.rs

```diff
@@ -411,6 +714,107 @@ fn eval(input: &Input, output: &mut Output, rule: &Rule) -> Result<Option<Value>
 
             Some(Value::Bool(a.iter().any(|x| b.contains(x))))
         }
+        Function::In(first_rule, second_rule) => {
+            let a = eval(input, output, first_rule)?.ok_or(Error::TypeError)?;
+            let b = eval(input, output, second_rule)?
+                .ok_or(Error::TypeError)?
+                .try_array()?;
+
+            Some(Value::Bool(b.contains(&a)))
+        }
+        Function::NotIn(first_rule, second_rule) => {
+            let a = eval(input, output, first_rule)?.ok_or(Error::TypeError)?;
+            let b = eval(input, output, second_rule)?
+                .ok_or(Error::TypeError)?
+                .try_array()?;
+
+            Some(Value::Bool(!b.contains(&a)))
+        }
+        Function::Between(first_rule, second_rule, third_rule) => {
```

Comment:
This should work the same way as `mulDiv` handling all numbers. Also a combination of rules can be `eval`-ed instead of writing the logic.